app/models/supporter.rb

```diff
@@ -66,6 +66,7 @@ class Supporter < ActiveRecord::Base
   has_many :activities, dependent: :destroy
   has_many :tickets
   has_many :recurring_donations
+  has_many :object_events, through: :nonprofit, source: :associated_object_events
```

Comment:
No, it should be connected to object_events like so:
I'd recommend following how it's done in Payout:

```suggestion
  has_many :object_events, as: :event_entity
```